### ISSUE TITLE
Infinite loop on discussions page

### ISSUE THREAD

**Issue by masvelio** [hicommonwealth/commonwealth#4853]
## Describe the bug

Not a big severity, because it seems that it is only reproducible on communities with few threads, but anyways, we should try to avoid infinite loops in the react components. In this case, the UI is freeze.

## To Reproduce

- go to discussions page of community that has few threads
- if everything works correctly, try to open thread and come back couple times
- on discussions page try to click some filters or `create thread` button
- actions do not response

https://github.com/hicommonwealth/commonwealth/assets/14819225/21aa6668-9108-439b-b424-6d69fe4a0abb

**In beta/production the console is clear, but locally, you can see a lot of warning/errors spitted out**

## Expected behavior

- UI should be functional

## Additional context

It needs investigation where exactly is a root cause of the infinite loop. Most likely it is a setState in the render method or some missing dependency in the useEffect

**Comment by alexyoung23j** [comment]
I'm unable to reproduce this.

**Comment by CowMuon** [comment]
@masvelio I see the loom is on your local. Are we able to reproduce this in any non-local environment? 

@alexyoung23j Where did you try to reproduce? 

Side note- are we still getting infinite loops / infinite redraws? I did think we were past that point, though not an objective determination, and more like we just haven't been seeing any of those recently

**Comment by masvelio** [comment]
The video is from local env but at that time I tested it also on beta and it was reproducible. However, I tested it again today and I am not able to reproduce it neither, so probably it was fixed by chance (or not by chance) by other PR. I am closing this ticket then.

Re side note @MuonShot We cannot associate the existence of infinite loops/rerenders solely with mithril or other hacks (e.g. event emmiters) that we used/use to make our React work properly. Infinite rerenders might happen in the most modern React code and the reason for this is developer's mistake or developer's unknowledge how react hooks work. Main reasons why infinite loops occur:
1. setting state conditionally in render phase instead of in useEffect
2. disobey `eslint-plugin-react-hooks` warnings about missing dependencies in the hooks' dependency arrays
